src/date-picker/index.ts

```diff
@@ -0,0 +1,202 @@
+import { html, PropertyValues } from "lit";
+import { property } from "lit/decorators.js";
+import { visiblePropConverter } from "../base/converter";
+import { formatDateByLocale, isValidDateFormat } from "../base/datetime/utils";
+import {
+  CustomEventDetail,
+  dispatchCustomEvent,
+  generateGUID,
+  KucBase
+} from "../base/kuc-base";
+import { validateProps } from "../base/validator";
+
+type DatePickerProps = {
+  className?: string;
+  error?: string;
+  id?: string;
+  label?: string;
+  disabled?: boolean;
+  requiredIcon?: boolean;
+  visible?: boolean;
+  language?: "ja" | "en" | "zh" | "auto";
+  value: string;
+};
+const Common_Language = "ja";
+export class DatePicker extends KucBase {
+  @property({ type: String, reflect: true, attribute: "class" }) className = "";
+  @property({ type: String, reflect: true, attribute: "id" }) id = "";
+  @property({ type: String }) error = "";
+  @property({ type: String }) label = "";
+  @property({ type: Boolean }) disabled = false;
+  @property({ type: Boolean }) requiredIcon = false;
+  @property({ type: String }) language = "auto";
+  @property({ type: String }) value = formatDateByLocale(
+    new Date().toDateString()
+  );
+  private _GUID: string;
+  @property({
+    type: Boolean,
+    attribute: "hidden",
+    reflect: true,
+    converter: visiblePropConverter
+  })
+  visible = true;
+
+  constructor(props?: DatePickerProps) {
+    super();
+    this._GUID = generateGUID();
+    const validProps = validateProps(props);
+    Object.assign(this, validProps);
+  }
+
+  update(changedProperties: PropertyValues) {
+    if (
+      changedProperties.has("value") &&
+      this.value &&
+      !isValidDateFormat(this.value, Common_Language)
+    ) {
+      this.value = "";
+    }
+    super.update(changedProperties);
+  }
+
+  private _getLanguage() {
+    // eslint-disable-next-line no-nested-ternary
```

Comment:
@h000780 
Could you please change logic for remove this eslint rule disabled?
Thank you!

```
private _getLanguage() {
  const langs = ["en", "ja", "zh"];
  if (langs.indexOf(this.language) !== -1) return this.language;
  
  if (langs !== "auto") return "en"; 
  
  if (langs.indexOf(document.documentElement.lang) !== -1) return document.documentElement.lang;

  return "en"; 
}
```
or
```
private _getLanguage() {
  let lang = this.language;
  if (lang === "auto") lang = document.documentElement.lang;

  if (["en", "ja", "zh"].indexOf(this.language) !== -1) return this.language;

  return "en"; 
}
```